### PR TITLE
17045 Fix back button on Breadcrumb shared component

### PR DESCRIPTION
**Pull request by leodube-aot**
*Issue #:* /bcgov/entity#17045

*Notes:*
The override causing the visual bug for the breadcrumb component comes from [overrides.scss](https://github.com/bcgov/namerequest/blob/feature-way-of-navigating/src/assets/styles/overrides.scss#L101), which overrides the `height` and `background-colour` of `v-btn`s. I am hesitant to remove the `!important` directives here since it is a global override and I am not sure what downstream effects that could have. Rather, I am moving the quick fix I had implemented for this bug in #618 (ticket [16633](https://app.zenhub.com/workspaces/entities-team-space-6143567664fb320019b81f39/issues/gh/bcgov/entity/16633)) to a more appropriate spot. Also, instead of targeting it using the sbc component id, I am using a custom class on the component to prevent any issues if the sbc id changes.

*Description of changes:*
- Move the custom breadcrumb button stylings into App.vue

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namerequest license (Apache 2.0).

### CODE REVIEW

File: src/App.vue
Comment (leodube-aot) [inline src/App.vue:49]: I would have liked to use an id here but it overrides the sbc assigned id and styles.

**Comment by leodube-aot** [comment]
@severinbeauvais Can you add some reviewers to this please? Thanks in advance

**Review by JazzarKarim — APPROVED** [review]
LGTM Leo 👍

File: src/App.vue
Comment (severinbeauvais) [inline src/App.vue:364]: Should this be in the shared component?

File: src/assets/styles/layout.scss
Comment (severinbeauvais) [inline src/assets/styles/layout.scss:24]: Oh, yeah, this doesn't belong here.

File: src/assets/styles/layout.scss
Comment (severinbeauvais) [inline src/assets/styles/layout.scss:30]: Wait - so this CSS broke the styling? But it fixes it if you put it in App.vue?

Could you explain why CSS properties got broken and by what?

File: src/App.vue
Comment (leodube-aot) [reply src/App.vue:364]: This is just to align it with the edge of the SBC Auth header. If the breadcrumb component will only ever be used underneath the sbc auth header then it can most likely be moved to the shared component.

File: src/assets/styles/layout.scss
Comment (leodube-aot) [reply src/assets/styles/layout.scss:30]: The two specific CSS properties that break it are in overrides.scss: [height: 2.5rem !important](https://github.com/bcgov/namerequest/blob/93f82f43983165d8661024fb50ebe4d56ddd9343/src/assets/styles/overrides.scss#L106) and [background-color: $app-blue !important;](https://github.com/bcgov/namerequest/blob/93f82f43983165d8661024fb50ebe4d56ddd9343/src/assets/styles/overrides.scss#L109). The css removed from here was the temporary fix used in [16633](https://github.com/bcgov/namerequest/pull/618) (also see [this discussion](https://github.com/bcgov/namerequest/pull/618/commits/f738f6235b0b4500c9a5a94afe27bae799584d49#r1253348992)). 

I believe the confusion is because when I created this ticket, I was not sure what the problem was but believed it was in the shared component. Since the issue was discovered to be specific to this repo, and the problem was technically already fixed, I think the scope of this ticket has changed to more of a clean-up ticket.

File: src/App.vue
Comment (severinbeauvais) [reply src/App.vue:364]: It looks to me like the Breadcrumb component looks OK in Storybook:
https://bcgov.github.io/bcrs-shared-components/?path=/story/component-breadcrumb--home-route

And I think it looks OK in the other UIs. Why is it different in Namerequest UI? Also I'd like to find another solution that doesn't require a "special number" in various places (here, auth-web, etc).

File: src/assets/styles/layout.scss
Comment (severinbeauvais) [reply src/assets/styles/layout.scss:30]: On the first point, I don't understand why having this CSS in layout.scss causes the problem, but moving it to App.vue fixes it.

One the second point, yes, I agree with you.

File: src/App.vue
Comment (leodube-aot) [reply src/App.vue:364]: I dug into this a bit and it seems other UIs do indeed rely on this "special number" to set the width of the breadcrumb. It isn't immediately obvious based on the css, but you can find it in[ business-filing-ui](https://github.com/bcgov/business-filings-ui/blob/bef1b9b0d48ca308447f6205f483998ca96a7e14/src/assets/styles/layout.scss#L35), [business-edit-ui](https://github.com/bcgov/business-edit-ui/blob/ef4a03ba0fc64f7b0297fd3aa46d9aaa39eadd18/src/assets/styles/layout.scss#L35), and [business-create-ui](https://github.com/bcgov/business-create-ui/blob/d0e45f28a223a1199201a7df29e513537d30fe75/src/assets/styles/layout.scss#L37). The reason it isn't obvious is that it is just being applied to the `.container` class, but when you inspect the breadcrumb you can see that this css gets applied to it (see screenshot). I agree that I don't like using an arbitrary number. I look into a different solution for this.

Screenshot (3)

File: src/assets/styles/layout.scss
Comment (leodube-aot) [reply src/assets/styles/layout.scss:30]: Hey Sev, just to clarify the first point, the css removed here is not the cause of the problem. If you look at namerequest ui on the feature branch you'll see the button looks fine even without the changes from this PR. (For example this temp url from Ketaki's PR when the environment get's fixed: https://namerequest-dev--pr-639-c9rqzr4l.web.app/).

This css is the solution, I just didn't like having it in the layout.css.

File: src/App.vue
Comment (severinbeauvais) [reply src/App.vue:364]: Shame on us 😆 

I accept the container width spec in the other UIs. I'd rather not see it nested in the Breadcrumb component.

Thanks for looking into this. It's not worth a ton of effort but I bet it will save some time in the future.

File: src/assets/styles/layout.scss
Comment (severinbeauvais) [reply src/assets/styles/layout.scss:30]: Oh, right! You fixed this previously. Now you're moving the CSS to where it's used. I like it. Resolving.

**Review by severinbeauvais — APPROVED** [review]
I'm assuming you'll set the container width somewhere appropriate, and when you do then you have my blessing to merge this :)

File: src/App.vue
Comment (leodube-aot) [reply src/App.vue:364]: Okay sounds good, so we'd like to move it back to layout.scss like in the other projects?

Comment (severinbeauvais) [reply src/App.vue:364]: Yes, but I'm assuming it's not specific to the Breadcrumbs component, right?

**Review by ketaki-deodhar — APPROVED** [review]

**Comment by ketaki-deodhar** [comment]
Please rebase before merging as I see package version conflicts

File: src/App.vue
Comment (leodube-aot) [reply src/App.vue]: Yeah that's correct, I have moved it back to layout.scss

**Comment by leodube-aot** [comment]
This PR should now be ready to merge (I don't have the permissions to do this).

**Review by severinbeauvais — APPROVED** [review]
LGTM

**Comment by severinbeauvais** [comment]
> This PR should now be ready to merge (I don't have the permissions to do this).

You do now. Please try again :)

**Pull request merged**
